### DISCUSSION TITLE
Does this library work with Storybook?

### DISCUSSION THREAD

**Discussion by takecchi** [kuma-ui/kuma-ui#264] [Q&A] [answered]
Is this library supported for use with [Storybook](https://storybook.js.org/)? If it is, please provide instructions on how to set it up.
I apologize if this information is already documented...

**Comment by ygkn** [comment, marked as answer]
I found a way to use Kuma UI in Storybook.

### With Webpack builder (Next.js)

If your project uses Webpack builder (or Next.js framework), you can use `"@kuma-ui/webpack-plugin"`.

```ts
// .storybook/main.ts

import KumaUIWebpackPlugin from "@kuma-ui/webpack-plugin";

// ...

const config: StorybookConfig = {
 // ...
 webpackFinal: (config) => {

   /**
    * @see https://github.com/kuma-ui/kuma-ui/blob/e276aea4dbd933800de5237d0411df9097451dfe/example/webpack/webpack.config.js
    */
   config.plugins = [...(config.plugins ?? []), new KumaUIWebpackPlugin({})];
   
   return config;
 },
 // ...
};

// ...
```

See [Working with Webpack plugins](https://storybook.js.org/docs/react/builders/webpack#working-with-webpack-plugins) Storybook docs for more details.

### With Vite builder


If you are using Vite builder in your project, it may work without configuration since Vite builder reads `vite.config.js`.

> The builder _will_ read your `vite.config.js` file, though it may change some of the options in order to work correctly.

If it does not work, try the following methods.

Use the `viteConfigPath` builder option to specify the config path, or use another config.

```ts
// .storybook/main.ts

const config = {
 framework: {
   name: '@storybook/react-vite',
   options: {
     builder: {
       viteConfigPath: '.storybook/customViteConfig.js',
     },
   },
 },
};
```

Use the `viteFinal` option to override the vite config.

```ts
// .storybook/main.ts

import { mergeConfig } from 'vite';
import KumaUI from "@kuma-ui/vite";

// ...

const config = {
 async viteFinal(config, { configType }) {
   return mergeConfig(config, {
     plugins: [
       KumaUI(),
     ],
   });
 },
};

// ...
```


See [Customize Vite config section of Storybook builder for Vite README](https://github.com/storybookjs/storybook/blob/next/code/builders/builder-vite/README.md#customize-vite-config)

---

This added to document. 
https://www.kuma-ui.com/docs/install#integrate-with-storybook

**Reply by takecchi** [reply]
Thank you for answering!
